src/lints/enum_tuple_variant_field_marked_deprecated.ron

```diff
@@ -0,0 +1,82 @@
+                        }
+
+                        variant {
+                            ... on TupleVariant {
+                                variant_name: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                deprecated @filter(op: "!=", value: ["$true"])
+
+                                field {
+                                    field_position: position @output @tag
+                                    public_api_eligible @filter(op: "=", value: ["$true"])
+                                    deprecated @filter(op: "!=", value: ["$true"])
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        # Filter out deprecated enums since rustdoc automatically marks their fields as deprecated.
+                        # This ensures we only detect fields that are explicitly marked with #[deprecated].
+                        deprecated @filter(op: "!=", value: ["$true"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        variant {
+                            ... on TupleVariant {
+                                name @filter(op: "=", value: ["%variant_name"])
+                                public_api_eligible @filter(op: "=", value: ["$true"])
+                                deprecated @filter(op: "!=", value: ["$true"])
+
+                                field {
+                                    position @filter(op: "=", value: ["%field_position"])
+                                    public_api_eligible @filter(op: "=", value: ["$true"])
+                                    deprecated @filter(op: "=", value: ["$true"])
+
+                                    span_: span @optional {
+                                        filename @output
+                                        begin_line @output
+                                        end_line @output
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+    },
+    error_message: "A field in an enum's tuple variant is now #[deprecated]. Downstream crates will get a compiler warning when accessing this field.",
+    per_result_error_template: Some("field at position {{field_position}} in enum variant {{enum_name}}::{{variant_name}} in {{span_filename}}:{{span_begin_line}}"),
```

Comment:
Any reason not to use the field name and Rust syntax for it? Like `MyEnum::Variant.0` for example?

Reply:
good call! I think I thought about this but yeah you are right this is more consistent with what we did for all other lints.